examples/VB_ASR/infer_constitutive_law_ASR_RCA.py

```diff
@@ -0,0 +1,286 @@
+        return 0.5 * np.dot(f, f), f @ diff
+    else:
+        f = me_vec(X)
+        return 0.5 * np.dot(f, f)
+
+
+###############################
+# VARIATIONAL BAYESIAN (VB)
+###############################
+
+prior_mvn = MVN(mean=X0, precision=np.diag([(abs(x)) ** (-2) for x in X0]))
+n0 = Gamma(shape=1.5, scale=175000.0)
+# n0 = Gamma(shape=0.84, scale=15000.)
+# n0 = Gamma(shape=0.5, scale=500000.)
+# n0 = Gamma(shape=0.84, scale=370000.)
+vb_results = vba(f=me_vec, x0=prior_mvn, noise0=n0, jac=return_jacobian)
+
+### Plot free energy
+plt.figure()
+n_vb = len(vb_results.free_energies)
+xs = list(range(1, n_vb + 1))
+plt.plot(xs, vb_results.free_energies, marker=".")
+plt.title(f"Free energy", size=12)
+plt.xlabel("VB iteration", size=12)
+plt.ylabel("F", size=12)
+plt.xticks(xs)
+plt.savefig(f"{_path}Free_energies.png", bbox_inches="tight", dpi=300)
+plt.show()
+
+### Plot Parameters
+from scipy.stats import norm, gamma
+
+pars_names = ["eps_inf", "Tau_c", "Tau_L"]
+pr_m = vb_results.param0.mean
+pr_std = vb_results.param0.std_diag
+post_m = vb_results.param.mean
+post_std = vb_results.param.std_diag
+noise0 = vb_results.noise0
+options_prior = {"linestyle": "", "color": "orange", "marker": "<", "fillstyle": "full"}
+options_posterior = {
+    "linestyle": "",
+    "color": "mediumblue",
+    "marker": ">",
+    "fillstyle": "full",
+}
+options_predict = {
+    "linestyle": "",
+    "color": "purple",
+    "marker": "d",
+    "fillstyle": "full",
+}
+for i, n in enumerate(pars_names):
+    pr_dist = norm(pr_m[i], pr_std[i])
+    post_dist = norm(post_m[i], post_std[i])
+    ## plot prior-posterior together
+    plot_pdfs(
+        [pr_dist, post_dist],
+        labels=["Prior", "Posterior"],
+        colors=[options_prior["color"], options_posterior["color"]],
+        tit=f"Parameter {n}",
+        xl=n,
+        _path=_path,
+        _name=f"Parameter_{n}",
+        _format=".png",
+        dpi=300,
+        sz=16,
+    )
+
+### Plot noises
+interval_prob = 0.99
+n0 = vb_results.noise0
+sh_pr = n0.shape
+sc_pr = n0.scale
+n1 = vb_results.noise
+sh_post = n1.shape
+sc_post = n1.scale
+g_pr = gamma(a=sh_pr, scale=sc_pr)
+g_post = gamma(a=sh_post, scale=sc_post)
+plot_pdfs(
+    [g_pr, g_post],
+    labels=["Prior", "Posterior"],
+    colors=[options_prior["color"], options_posterior["color"]],
+    interval_prob=interval_prob,
+    tit=f"Inference of precision of noise",
+    xl="$\phi$",
+    _path=_path,
+    _name=f"Noise_precision",
+    _format=".png",
+    dpi=300,
+    sz=16,
+)
+
+
+if return_jacobian:
+    # propagation of parameters distributions into the model
+    f_at_posterior, diff = eps_total(
+        ts_data, eps_inf=post_m[0], Tau_c=post_m[1], Tau_L=post_m[2]
+    )
+    cov = vb_results.param.cov
+    cov_propagated = gaussian_error_propagator(
+        jac=diff, stds=post_std, cov=cov, rtol=1e-12, atol=1e-5
+    )
+    # uncertainty of the identified noise
+    cov_inv = vb_results.cov_inv[0]

```

Comment:
Not sure I understand the structure of the data that vb_result returns

Reply:
This cov_inv is the inverse of covariance in the likelihood (scaled by Phi as a scalar latent parameter). In most cases including the example added, this matrix is just identity matrix. However, the computation of uncertainty of noise model is done for the general case (cov_inv being a certain matrix other than identity).